sw/device/silicon_creator/manuf/lib/sram_exec_test.c

```diff
@@ -0,0 +1,69 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdint.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/pinmux_testutils.h"
+#include "sw/device/lib/testing/test_framework/status.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_uart_t uart0;
+static dif_pinmux_t pinmux;
+static dif_otp_ctrl_t otp;
+
+static const uint8_t kNumDeviceId = 8;
+
+status_t sram_main() {
+  if (kDeviceType != kDeviceSimDV) {
+    // Configure the pinmux.
+    TRY(dif_pinmux_init(
+        mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+    pinmux_testutils_init(&pinmux);
+
+    // Initialize UART.
+    TRY(dif_uart_init(
+        mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart0));
+    TRY(
+        dif_uart_configure(&uart0, (dif_uart_config_t){
+                                       .baudrate = kUartBaudrate,
+                                       .clk_freq_hz = kClockFreqPeripheralHz,
+                                       .parity_enable = kDifToggleDisabled,
+                                       .parity = kDifUartParityEven,
+                                       .tx_enable = kDifToggleEnabled,
+                                       .rx_enable = kDifToggleEnabled,
+                                   }));
+    base_uart_stdout(&uart0);
+  }
+  LOG_INFO("Initializing OTP...");
+  TRY(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp));
+
+  dif_otp_ctrl_config_t config = {
+      .check_timeout = 100000,
+      .integrity_period_mask = 0x3ffff,
+      .consistency_period_mask = 0x3ffffff,
+  };
+  TRY(dif_otp_ctrl_configure(&otp, config));
+
+  uint32_t otp_device_id;
+  for (int i = 0; i < kNumDeviceId; i++) {
+    TRY(otp_ctrl_testutils_dai_read32(&otp, kDifOtpCtrlPartitionHwCfg, i * 4,
+                         &otp_device_id));
+
+    LOG_INFO("Device_ID_%d = %08x", i, otp_device_id);
+  }
+
+  LOG_INFO("Hello from SRAM");
+
+  return OK_STATUS(42);
```

Comment:
Is there a reason for the "42" return code?

Reply:
Not it's just to debugging purposes, this will go away when the test is done properly.